numpy/lib/function_base.py

```diff
@@ -309,7 +309,8 @@ def average(a, axis=None, weights=None, returned=False):
         return a tuple with the average as the first element and the sum
         of the weights as the second element. The return type is `Float`
         if `a` is of integer type, otherwise it is of the same type as `a`.
-        `sum_of_weights` is of the same type as `average`.
+        `sum_of_weights` is of the same type as `average`. The return type
+	    is same as the data type of `weights` if provided.
```

Comment:
This is not really true either. `average(float32_arr, int32_weights)` returns a float64 result

Reply:
@eric-wieser : 
Shall I reframe it to something like : The return type is same as that of weights if provided in `float` variant?

I tested it out after your review and apparently only the case (as mentioned in the issue) is the exception otherwise it consistently gives `float64`

Reply:
You could probably say something like the "common type" of both, the type is the same as operations like `average + weights`.